tests/apps/courses/test_templates_course_detail_opengraph.py

```diff
@@ -185,3 +187,32 @@ def test_open_graph_course_description_no_introduction(self):
         response = self.client.get(url)
 
         self.assertNotContains(response, "og:description")
+
+    @mock.patch.object(
+        FileSystemStorage,
+        "url",
+        lambda storage_instance, name: f"https://test-bucket.s3.amazonaws.com/{name:s}",
+    )
+    def test_open_graph_course_meta_og_image_media_storage_s3(self):
```

Comment:
I would say you are not testing the problem you had which was that it was breaking when MEDIA_URL was not set?

Reply:
If you want to be sure that it works with Boto3, the only way is to install it in dev dependencies and test with it and forcing the MEDIA_URL setting to None or something like this?

Reply:
@sampaccoud  the problem is that s3boto3 django-storages implementation don't use the `MEDIA_URL` setting. 
The `MEDIA_URL` setting isn't used by [s3boto3.py](https://github.com/jschneier/django-storages/blob/master/storages/backends/s3boto3.py), that is why our tests didn't detect the problem.

To make the richie implementation and its tests more generic, to run on any kind of storage, we shouldn't use or run tests for a specific storage solution, but instead use the `django.core.files.storage.Storage` base class /interface. That is why I mocked the `url` method. We could replace `s3.amazonaws.com` on the tests with any think of domain.

Reply:
ok... then let's derive it in 3 tests for each kind of domain?

Reply:
Yes.
* One for full URL
* another for an URL that starts with a double slash
* and a last one for an URL start with a relative URL.

Reply:
:+1: